xds/src/main/java/io/grpc/xds/sds/TlsCertificateStore.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2019 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.xds.sds;
+
+import com.google.common.base.Preconditions;
+import com.google.protobuf.ByteString;
+import io.envoyproxy.envoy.api.v2.auth.TlsCertificate;
+import io.envoyproxy.envoy.api.v2.core.DataSource;
+import io.grpc.Internal;
+import java.io.InputStream;
+
+/**
+ * TlsCertificate's PrivateKey and Certificate are extracted into InputStream's.
+ * This is used by the gRPC SSLContext/Protocol Negotiator and is internal
+ */
+@Internal
+public final class TlsCertificateStore {
+
+  private final InputStream privateKeyStream;
+  private final InputStream certChainStream;
+
+  private InputStream getInputStreamFromDataSource(DataSource dataSource) {
+    Preconditions.checkNotNull(dataSource);
+    ByteString dataSourceByteString = null;
+    if (dataSource.getSpecifierCase() == DataSource.SpecifierCase.INLINE_BYTES) {
+      dataSourceByteString = dataSource.getInlineBytes();
+    } else if (dataSource.getSpecifierCase() == DataSource.SpecifierCase.INLINE_STRING) {
+      dataSourceByteString = dataSource.getInlineStringBytes();
+    } else {
+      throw new IllegalArgumentException(
+          "dataSource of type " + dataSource.getSpecifierCase() + " not supported");
+    }
+    return dataSourceByteString.newInput();
+  }
+
+  /**
+   * Creates the Store out of the TlsCertificate object of xDS.
+   *
+   * @param tlsCertificate  TlsCertificate Object of xDS
+   */
+  public TlsCertificateStore(TlsCertificate tlsCertificate) {
```

Comment:
Is this constructor only used for testing?

Reply:
It will be used in future (another PR) with actual SDS responses returning TlsCertificate objects. Do you recommend I remove it now if not used?

Reply:
I would keep it although I prefer to reuse the other constructor by
```
this(
    getInputStreamFromDataSource(checkNotNull(tlsCertificate).getPrivateKey()),
    getInputStreamFromDataSource(tlsCertificate.getCertificateChain());
```

Reply:
I know Java evaluation order implies checkNotNull will happen first although the general recommendation is to not ["crucially rely on it"](https://docs.oracle.com/javase/specs/jls/se8/html/jls-15.html#jls-15.7). I will incorporate this suggestion though